samples/CMakeLists.txt

```diff
@@ -12,6 +12,9 @@ else()
     set(LINK_TCOD libtcod::libtcod)
     set(SDL2MAIN SDL2::SDL2main)
 endif()
+if(MINGW)
+    set(LINK_TCOD mingw32 "${LINK_TCOD}")
+endif()
```

Comment:
I don't understand where this fix comes from.  Why is `mingw32` hard coded here?

Reply:
Because when compiling SDL2 with MinGW in the MSYS2 environment, you need to link `libmingw32.a` first, or else the linker stage will fail. This is a widely-known issue.